samples/application-insights/sample.md

```diff
@@ -0,0 +1,82 @@
+
+snippet: load-simulator
+
+
+## Connect to Azure Application Insights
+
+To connect the sample code to Application Insights, the instrumentation key needs to be provided.
+
+By default the sample code loads this key from an environment variable called `ApplicationInsightKey`. Either set this environment variable or paste the instrumentation key in the following section.
+
+snippet: configure-ai-instrumentation-key
+
+## Use NServiceBus.Metrics to capture metric values
+
+To capture the metric values generated by NServiceBus, a class called `ApplicationInsightsFeature` is added to the sample. It is responsible for enabling the NServiceBus metrics feature.
+
+snippet: enable-nsb-metrics
+
+It ensures that the telemetry probe is subscribed to the event stream. During this stage the `ApplicationInsightsProbeCollector` is configured and wired. Information such as *endpoint name* and *input queue* are passed to allow advanced data analysis in Application Insights.  
+
+snippet: register-probe
+
+Events and metrics are buffered and the Application Insights telemetry client does this every minute. If the endpoint instance would gracefully stop it needs to ensure that any pending events are flushed to Application Insight.
+
+snippet: flush-probe
+
+## Send captured metric data to Application Insights
+
+The `ApplicationInsightsProbeCollector` is used to pass NServiceBus captured metric data to the Application Insights instance configured earlier. Information passed is added as telemetry context properties, which will be attached to every metric (NServiceBus signal probe) and event (NServiceBus duration probe) captured via the telemetry client. The telemetry client is used to communicate with Application Insights.
+
+snippet: telemetry-client
+
+When the `ApplicationInsightsProbeCollector` is registered with the NServiceBus metrics API, it registers observer callbacks to receive probe names, signals and durations. Signals are converted into Application Insights custom events. Durations are converted into Application Insights metrics.
+
+snippet: observers-registration
+
+
+## Setting up charts for metrics and events (signals and durations)
+
+The easiest way to setup graphs for metrics in Application Insights is to go to the `Metrics Explorer`, add a new graph selecting the custom metric of choice and then use the push pin symbol to add it to a dashboard.
+
+The following graphs show the following metrics:
+
+ * Maximum critical time & critical time grouped by instance id.
+ * Average processing time, not grouped.
+ * Minimum hours countdown until Service Level Agreement violation.
+
+<img alt="example graph metrics" src="example-graph-metrics.png" width="50%" />
+
+Similar for events, the following graphs shows all events grouped by event name. This graph can be furthered filtered down to e.g. show failures only.
+
+<img alt="example graph events" src="example-graph-events.png" width="50%" />
```

Comment:
Indeed, that works even better because on 4K these images now render great in preview. Before they were quite small when the window was not taking the whole screen.